source/common/quic/envoy_quic_server_stream.cc

```diff
@@ -475,8 +475,10 @@ bool EnvoyQuicServerStream::hasPendingData() {
   return (!write_side_closed()) && BufferedDataBytes() > 0;
 }
 
+// TODO(https://github.com/envoyproxy/envoy/issues/23564): enable HTTP Datagram support by looking
+// at the request/response headers for CONNECT-UDP support.
```

Comment:
This makes me wonder if this method should be protected?

Reply:
I'm not quite sure how the next PR works :) If that PR calls this method from within this class, then I'd say protected would be the way to go. But if that PR needs it to be public, I'm happy for it to stay this way.

Reply:
Ah, got it. The plan is to call this in a different class so I will leave it as is. Done.